nbconvert/filters/markdown_mistune.py

```diff
@@ -29,6 +35,31 @@ class MathInlineGrammar(mistune.InlineGrammar):
     text = re.compile(r'^[\s\S]+?(?=[\\<!\[_*`~$]|https?://| {2,}\n|$)')
 
 
+class MathBlockLexer(mistune.BlockLexer):
+    default_rules = (['block_math', 'latex_environment']
+                     + mistune.BlockLexer.default_rules)
+
+    def __init__(self, rules=None, **kwargs):
+        if rules is None:
+            rules = MathBlockGrammar()
+        super(MathBlockLexer, self).__init__(rules, **kwargs)
+
+    def parse_block_math(self, m):
```

Comment:
Change this to have the correct name for what you decide to modify the main regex to. In the case of `multiline_math` it would be `parse_multiline_math`.